frame/staking/src/lib.rs

```diff
@@ -1646,19 +1638,31 @@ impl<T: Trait> Module<T> {
 	}
 }
 
-impl<T: Trait> pallet_session::OnSessionEnding<T::AccountId> for Module<T> {
-	fn on_session_ending(_ending: SessionIndex, start_session: SessionIndex) -> Option<Vec<T::AccountId>> {
+impl<T: Trait> pallet_session::SessionManager<T::AccountId> for Module<T> {
+	fn new_session(new_index: SessionIndex) -> Option<Vec<T::AccountId>> {
 		Self::ensure_storage_upgraded();
-		Self::new_session(start_session - 1).map(|(new, _old)| new)
+		if new_index == 0 {
+			return <Module<T>>::select_validators().1
+		}
+		Self::new_session(new_index - 1)
```

Comment:
This new session will trigger the reward for the previous session and create a new one, this is wrong and is intented to be fixed by https://github.com/paritytech/substrate/pull/4474
This PR doesn't the behavior of staking module